receipt_chroma/receipt_chroma/s3/helpers.py

```
@@ -503,3 +844,326 @@ def upload_delta_tarball(
+                    if member_path.startswith("..") or member_path.startswith("/"):
+                        logger.warning(
+                            "Skipping potentially unsafe path: %s", member.name
+                        )
+                        continue
+
+                    if member.isfile():
+                        file_count += 1
+                        total_size += member.size
+
+                # Extract all (after validation)
+                tar.extractall(local_path)
```

Ruff (0.14.10)

Uses of `tarfile.extractall()`

(S202)